File: app/src/main/java/com/pandatone/kumiwake/sekigime/SquareTableCustom.java
package com.pandatone.kumiwake.sekigime;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.CompoundButton;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.SeekBar;
import android.widget.TextView;

import com.pandatone.kumiwake.R;
import com.pandatone.kumiwake.member.Name;

import java.util.ArrayList;

import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by atsushi_2 on 2016/07/15.
 */
public class SquareTableCustom extends AppCompatActivity {
    RadioGroup squareRadioGroup;
    RadioButton doubleDeployButton;
    SeekBar seekBar;
    TextView noText;
    int seatNo = 0, mingroupNo = 1000;
    boolean doubleDeploy;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.square_custom);
        ButterKnife.bind(this);
        squareRadioGroup = (RadioGroup) findViewById(R.id.squareGroup);
        doubleDeployButton = (RadioButton) findViewById(R.id.doubleSquareType);
        seekBar = (SeekBar) findViewById(R.id.square_seek_bar);
        noText = (TextView) findViewById(R.id.seat_number);

        if (SekigimeResult.Normalmode) {
            ArrayList<ArrayList<Name>> array = SekigimeResult.arrayArrayNormal;
            for (int i = 0; i < array.size(); i++) {
                if (array.get(i).size() < mingroupNo) {
                    mingroupNo = array.get(i).size();
                }
            }
        } else {
            ArrayList<ArrayList<String>> array = SekigimeResult.arrayArrayQuick;
            for (int i = 0; i < array.size(); i++) {
                if (array.get(i).size() < mingroupNo) {
                    mingroupNo = array.get(i).size();
                }
            }
        }
        seekBar.setMax(mingroupNo / 4);
        if (mingroupNo < 4) {
            seekBar.setEnabled(false);
        }
        doubleDeployButton.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (doubleDeployButton.isChecked() == true) {
                    seekBar.setMax(mingroupNo / 4);
                    if (mingroupNo<4) {
                        seekBar.setEnabled(false);
                    }
                } else {
                    seekBar.setMax(mingroupNo / 3);
                    if (mingroupNo<3) {
                        seekBar.setEnabled(false);
                    }else {
                        seekBar.setEnabled(true);
                    }
                }

            }
        });


        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromTouch) {
                seatNo = progress;
                noText.setText(String.valueOf(seatNo));
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });
    }

    @OnClick(R.id.move_result)
    void onClicked() {
        SekigimeResult.square_no = seatNo;
        doubleDeploy = doubleDeployButton.isChecked();
        SekigimeResult.doubleDeploy = doubleDeploy;
        Intent intent = new Intent(getApplicationContext(), SekigimeResult.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/pandatone/kumiwake/sekigime/SquareTableCustom.kt
package com.pandatone.kumiwake.sekigime

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.SeekBar
import androidx.appcompat.app.AppCompatActivity
import com.pandatone.kumiwake.R
import com.pandatone.kumiwake.databinding.SquareCustomBinding
import kotlin.math.ceil

/**
 * Created by atsushi_2 on 2016/07/15.
 */
class SquareTableCustom : AppCompatActivity() {
    private lateinit var binding: SquareCustomBinding

    private var seatNo = 0
    private var minGroupNo = 1000
    private var doubleDeploy: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = SquareCustomBinding.inflate(layoutInflater)
        setContentView(binding.root)
        findViewById<Button>(R.id.move_result).setOnClickListener { onNext() }

        val array = SekigimeResult.teamArray
        for (item in array) {
            val size = item.size
            if (size < minGroupNo) {
                minGroupNo = size
            }
        }

        binding.squareSeekBar.max = minGroupNo / 4
        if (minGroupNo < 4) {
            binding.squareSeekBar.isEnabled = false
        }
        binding.doubleSquareType.setOnCheckedChangeListener { _, _ ->
            if (binding.doubleSquareType.isChecked) {
                binding.squareSeekBar.max = minGroupNo / 4
                if (minGroupNo < 4) {
                    binding.squareSeekBar.isEnabled = false
                }
            } else {
                binding.squareSeekBar.max = ceil(minGroupNo / 3.toDouble()).toInt()
                binding.squareSeekBar.isEnabled = minGroupNo >= 3
            }
        }


        binding.squareSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onStartTrackingTouch(squareSeekBar: SeekBar) {}

            override fun onProgressChanged(
                squareSeekBar: SeekBar,
                progress: Int,
                fromTouch: Boolean
            ) {
                seatNo = progress
                binding.seatNumber.text = seatNo.toString()
            }

            override fun onStopTrackingTouch(squareSeekBar: SeekBar) {}
        })
    }

    private fun onNext() {
        SekigimeResult.square_no = seatNo
        doubleDeploy = binding.doubleSquareType.isChecked
        SekigimeResult.doubleDeploy = doubleDeploy
        val intent = Intent(applicationContext, SekigimeResult::class.java)
        startActivity(intent)
    }
}
